### PR TITLE
File based file name storage backend implementation

### PR DESCRIPTION
**Pull request by slashben**

### CODE REVIEW

File: pkg/filehandler/v1/file_handler.go [hunk 1]
```diff
@@ -56,6 +61,16 @@ func (b BoltFileHandler) GetFiles(ctx context.Context, container string) (map[st
...
+
+	// Purge the bucket
+	err = b.fileDB.Update(func(tx *bolt.Tx) error {
+		err := tx.DeleteBucket([]byte(container))
+		if err != nil {
+			return fmt.Errorf("delete bucket: %s", err)
+		}
+		return nil
+	})
+
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:73]: This must be deleted

File: pkg/filehandler/v1/file_handler.go [hunk 2]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
 		return nil
 	})
 }
+
+type File struct {
+	// File object
+	FileObject *os.File
+	// Bufio writer
+	Writer *bufio.Writer
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:94]: Remove this as it is not needed

File: pkg/filehandler/v1/file_handler.go [hunk 3]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
...
+	defer os.Remove(storageFileCopyName)
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		fileList[scanner.Text()] = true
+	}
+	if err := scanner.Err(); err != nil {
+		return fileList, err
+	}
+	return fileList, nil
+}
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:208]: Need to protect this as per bucket against re-entrancy (there is a single "copy" file that can be overwritten)

File: pkg/filehandler/v1/file_handler.go [hunk 4]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
...
+func (f *FsFileHandler) GetFiles(ctx context.Context, container string) (map[string]bool, error) {
+	// Normalize the bucket name
+	container = NormalizeBucketName(container)
+
+	// Create a map of string to bool
+	fileList := make(map[string]bool)
+	// Check if bucket exists in the map
+	f.Lock.Lock()
+	fileObject, ok := f.fileMap[container]
+	if !ok {
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:158]: Move here the lock

File: pkg/filehandler/v1/file_handler.go [hunk 5]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
...
+
+	// Do not use defer here! It makes life much harder
+
+	// Copy the file to a new file
+	storageFileName := fmt.Sprintf("%s/%s.txt", f.dirPath, container)
+	storageFileCopyName := fmt.Sprintf("%s/%s_copy.txt", f.dirPath, container)
+	// Copy the file
+	err := copyFile(storageFileName, storageFileCopyName)
+	if err != nil {
+		fileObject.FileObject, _ = os.OpenFile(storageFileName, os.O_CREATE|os.O_WRONLY, 0644)
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:177]: Open to append

File: pkg/filehandler/v1/file_handler.go [hunk 6]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
...
+		fileObject.Lock.Unlock()
+		return fileList, err
+	}
+
+	// Open the file
+	fileObject.FileObject, err = os.OpenFile(storageFileName, os.O_CREATE|os.O_WRONLY, 0644)
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:183]: Move unlock here

File: pkg/filehandler/v1/file_handler.go [hunk 7]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
...
+		fileObject.Lock.Unlock()
+		return fileList, err
+	}
+	fileObject.Writer = bufio.NewWriter(fileObject.FileObject)
+
+	// Unlock the file and let other goroutines use it
+	fileObject.Lock.Unlock()
+
+	// Read the file line by line
+	file, err := os.Open(storageFileCopyName)
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:194]: Generate temp filename generator

File: pkg/filehandler/v1/file_handler.go [hunk 8]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
...
+	if !ok {
+		return fmt.Errorf("bucket does not exist for container %s", bucket)
+	}
+
+	// Lock the file
+	fileObject.Lock.Lock()
+	// Close the file
+	fileObject.FileObject.Close()
+	// Remove the file
+	storageFileName := fmt.Sprintf("%s/%s.txt", f.dirPath, bucket)
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:227]: Generate filename with a function

File: pkg/filehandler/v1/file_handler.go [hunk 9]
```diff
@@ -71,3 +86,183 @@ func (b BoltFileHandler) RemoveBucket(ctx context.Context, bucket string) error
...
+	err := os.Remove(storageFileName)
+	if err != nil {
+		fileObject.Lock.Unlock()
```
Comment (slashben) [inline pkg/filehandler/v1/file_handler.go:230]: Move unlock here

**Review by slashben — COMMENTED** [review]
Fix it

**Comment by dwertent** [comment]
We have decided to move forward with the in-memory storage.

**Pull request closed without merging**